### PR TITLE
Make scope name unique and add missing header.

### PR DESCRIPTION
**Pull request by yang-g**
The scope name issue is exposed because we have two places in chttp2_transport.cc where two GPR_TIMER_SCOPE's (with different tags) are defined in the same code block.

The missing header is caused by recent PR #14676. Adding the header to a source file seems to be fine according to the PR.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by yang-g** [comment]
We cannot use grpc_core::New because that will call into timer library in a circular way. Changed back to pthread. Tested locally.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/lib/profiling/basic_timers.cc [hunk 1]
```diff
@@ -68,7 +68,7 @@ static pthread_cond_t g_cv;
 static gpr_timer_log_list g_in_progress_logs;
 static gpr_timer_log_list g_done_logs;
 static int g_shutdown;
-static grpc_core::Thread* g_writing_thread;
+static pthread_t g_writing_thread;
```
Comment (sreecha) [inline src/core/lib/profiling/basic_timers.cc]: Why does this have to be `pthread_t` ?

**Comment by yang-g** [comment]
I added a comment for this but did not make it clear I guess. When basic_timers tries to create a grpc_core::Thread, the ctor needs to call gpr_malloc internally and that triggers another call into basic_timers, which will cause a deadlock. That is the reason in the file we have been using pthread types rather than gpr types.

**Comment by sreecha** [comment]
Sorry Yang..I didn't see the previous comment "We cannot use grpc_core::New because that will call into timer library in a circular way. Changed back to pthread. Tested locally" - it got lost in the noise..

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**